File: training/javascript/src/de/sots/cellarsandcentaurs/application/CreatureUtil.js
import * as model from "../domain/model/Creature.js";
import { Fightable } from "../domain/model/Fightable.js";

export class CreatureUtil {
    static STANDARD_ARMOR_CLASS_DESCRIPTION = "Natural Armor";

    /*
     * Counts the treasure hoard a creature guards.
     */
    static countHoard(creature) {
        return creature instanceof model.Creature ? creature.getHitPoints().max * 10 : 0;
    }
}

File: sample-projects/java/src/main/java/de/sots/cellarsandcentaurs/application/CreatureUtil.java
package de.sots.cellarsandcentaurs.application;

import de.sots.cellarsandcentaurs.domain.model.*;
import de.sots.cellarsandcentaurs.domain.model.Fightable;

public final class CreatureUtil {
    public static final String STANDARD_ARMOR_CLASS_DESCRIPTION = "Natural Armor";
    private static final int GOLD_PER_HIT_POINT = 3;

    private CreatureUtil() {
    }

    /*
     * Counts the treasure hoard a creature guards.
     */
    public static int treasureHoard(Creature creature) {
        HitPoints hitPoints = creature.getHitPoints();
        return hitPoints.getMax() * GOLD_PER_HIT_POINT;
    }

    public static boolean isWounded(Creature creature) {
        return creature.getHitPoints().getCurrent() < creature.getHitPoints().getMax();
    }
}
